core/src/main/java/io/aiven/klaw/validation/TopicRequestValidatorImpl.java

```diff
@@ -212,58 +216,71 @@ private boolean checkIfPromotionOfTopic(
+            return false;
           }
-        }
-      }
-    } catch (Exception e) {
-      log.error("Unable to set topic partitions, setting default from properties.", e);
-    }
 
-    try {
-      if (topicPrefix != null
-          && !topicPrefix.isBlank()
-          && !topicRequestReq.getTopicname().startsWith(topicPrefix)) {
-        log.error(
-            "Topic prefix {} does not match. {}", topicPrefix, topicRequestReq.getTopicname());
-        updateConstraint(
-            constraintValidatorContext,
-            String.format(TOPICS_VLD_ERR_115, topicRequestReq.getTopicname()));
-        return false;
-      }
+          if (topicSuffix != null
+              && !topicSuffix.isBlank()
+              && !topicRequestReq.getTopicname().endsWith(topicSuffix)) {
+            log.error(
+                "Topic suffix {} does not match. {}", topicSuffix, topicRequestReq.getTopicname());
+            updateConstraint(
+                constraintValidatorContext,
+                String.format(TOPICS_VLD_ERR_116, topicRequestReq.getTopicname()));
+            return false;
+          }
+        } else {
+          if (topicRegex != null
+              && !topicRegex.isBlank()
+              && !isRegexAMatch(topicRequestReq, topicRegex)) {
 
-      if (topicSuffix != null
-          && !topicSuffix.isBlank()
-          && !topicRequestReq.getTopicname().endsWith(topicSuffix)) {
-        log.error(
-            "Topic suffix {} does not match. {}", topicSuffix, topicRequestReq.getTopicname());
-        updateConstraint(
-            constraintValidatorContext,
-            String.format(TOPICS_VLD_ERR_116, topicRequestReq.getTopicname()));
+            log.error(
+                "Topic Regex {} does not match. {}", topicRegex, topicRequestReq.getTopicname());
+            updateConstraint(
+                constraintValidatorContext,
+                String.format(TOPICS_VLD_ERR_118, topicRequestReq.getTopicname()));
+            return false;
+          }
+        }
+
+      } catch (Exception e) {
+        log.error("Unable to set topic partitions, setting default from properties.", e);
+        updateConstraint(constraintValidatorContext, TOPICS_VLD_ERR_117);
         return false;
       }
-    } catch (Exception e) {
-      log.error("Unable to set topic partitions, setting default from properties.", e);
-      updateConstraint(constraintValidatorContext, TOPICS_VLD_ERR_117);
-      return false;
     }
-
     return true;
   }
 
+  // TODO Review rej and see if this would provide a better experience for Klaw.
+  private boolean isRegexAMatch(TopicRequestModel topicRequestReq, String topicRegex) {
+    Pattern p = Pattern.compile(topicRegex);
+    Matcher m = p.matcher(topicRequestReq.getTopicname());
+    return m.matches();
+  }
+
+  private static String getValueOrDefault(List<String> params, String defaultValue) {
+    return (params != null && params.size() > 0) ? params.get(0) : defaultValue;
```

Comment:
Its always first element ?, either we can remove list

Reply:
Yes, if we remove the List it simplifies it a lot. 
The only lists I think we really need is for the partition and replication lists, the defaultPartionSize and defaultReplicationSize could also be made Strings imo